File: src/main/java/no/nav/familie/dokgen/configuration/DelayedShutdownHook.java
package no.nav.familie.dokgen.configuration;

import org.springframework.context.ConfigurableApplicationContext;

public class DelayedShutdownHook extends Thread {

    private final ConfigurableApplicationContext applicationContext;

    public DelayedShutdownHook(final ConfigurableApplicationContext applicationContext) {

        this.applicationContext = applicationContext;
    }

    @Override
    public void run() {
        try {
            // https://github.com/kubernetes/kubernetes/issues/64510
            // https://nav-it.slack.com/archives/C5KUST8N6/p1543497847341300
            Thread.sleep(5000L);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        applicationContext.close();
        super.run();
    }
}

File: src/main/kotlin/no/nav/dokgen/configuration/DelayedShutdownHook.kt
package no.nav.dokgen.configuration

import no.nav.dokgen.services.TemplateService
import org.slf4j.LoggerFactory
import org.springframework.context.ConfigurableApplicationContext

class DelayedShutdownHook(private val applicationContext: ConfigurableApplicationContext) : Thread() {
    override fun run() {
        try {
            // https://github.com/kubernetes/kubernetes/issues/64510
            // https://nav-it.slack.com/archives/C5KUST8N6/p1543497847341300
            sleep(5000L)
        } catch (e: InterruptedException) {
            LOG.error("En feil med shutdown oppstod", e)
        }
        applicationContext.close()
        super.run()
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(DelayedShutdownHook::class.java)
    }
}